src/path.util.ts

```diff
@@ -26,8 +28,10 @@ export const toDataPath = (path: string): string =>
 export const resolveSchema = (schema: JsonSchema, path: string): JsonSchema => {
   const validPathSegments = path.split('/');
   const resultSchema = validPathSegments.reduce((curSchema, pathSegment) =>
-      pathSegment === '#' ? curSchema : curSchema[pathSegment], schema);
-  if (resultSchema.$ref !== undefined) {
+      pathSegment === '#' || pathSegment === undefined || pathSegment === '' ?
```

Comment:
The `pathSegment === '#' || pathSegment === undefined || pathSegment === ''` condition